### PR TITLE
enhancement: update how to build docker images

### PR DESCRIPTION
**Pull request by kneal**
Listed changes for plugin docs:

* Removed content from Docker, and point to Kaniko
* Added new section for Kaniko docs
* Added new section for Makisu docs

Listed changes for usage docs:

* Add content for how to build docker images within Vela

### CODE REVIEW

File: content/plugins/pipeline/registry/kaniko.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+title: "Kaniko"
+---
+
+## Overview
+
+This plugin uses [Kaniko](https://github.com/GoogleContainerTools/kaniko) to provide the ability to build and publish [Docker](https://www.docker.com/) images in a Vela pipeline without evaluated access on the workers.
```
Comment (jbrockopp) [inline content/plugins/pipeline/registry/kaniko.md]:
```suggestion
This plugin uses [Kaniko](https://github.com/GoogleContainerTools/kaniko) to provide the ability to build and publish [Docker](https://www.docker.com/) images in a Vela pipeline without elevated access on the workers.
```

File: content/plugins/pipeline/registry/makisu.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
+---
+title: "Makisu"
+---
+
+## Description
+
+This plugin uses [Kaniko](https://github.com/uber/makisu) to provide the ability to build and publish [Docker](https://www.docker.com/) images in a Vela pipeline without evaluated access on the workers.
```
Comment (jbrockopp) [inline content/plugins/pipeline/registry/makisu.md]:
```suggestion
This plugin uses [Kaniko](https://github.com/uber/makisu) to provide the ability to build and publish [Docker](https://www.docker.com/) images in a Vela pipeline without elevated access on the workers.
```

File: content/usage/getting-started/docker.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+---
+
+We assume you understand how to build and run Docker images. If you need assistance on how to get started with Docker we recommend you see their documentation for [getting set up](https://docs.docker.com/get-started/).
+
+Vela runs all workloads within Docker containers. Which essentially gives us two core different ways to build Docker images:
+
+* Without elevated daemon access
+* With elevated daemon access
+
+Both options have disadvantages and advantages, so we encourage all Vela administrators to way the pros/cons of how they want to build Docker images for their cluster.
```
Comment (jbrockopp) [inline content/usage/getting-started/docker.md]:
```suggestion
Both options have disadvantages and advantages, so we encourage all Vela administrators to weigh the pros/cons of how they want to build Docker images for their cluster.
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

**Review by jbrockopp — DISMISSED** [review]
LGTM

File: content/plugins/pipeline/registry/docker.md [hunk 4]
```diff
@@ -4,228 +4,40 @@ title: "Docker"
 
 ## Overview
 
-This plugin enables the ability to build and publish [Docker](https://www.docker.com/) images in a Vela pipeline.
+{{% alert color="warning" %}}
+This plugin was originally used to build **all** Docker images. The functionality of this plugin still exists within [Kaniko](/kaniko)
```
Comment (wass3r) [inline content/plugins/pipeline/registry/docker.md]: is that the right link to kaniko?

File: content/plugins/pipeline/registry/makisu.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+| `repo`          | name of the repository for the image                               | `true`   | `N/A`             |
+| `username`      | user name for communication with the registry                      | `true`   | `N/A`             |
+
+## Template
+
+COMING SOON!
+
+## Troubleshooting
+
+Below are a list of common problems and how to solve them:
```
Comment (wass3r) [inline content/plugins/pipeline/registry/makisu.md:210]: change to "COMING SOON!"?

File: content/plugins/pipeline/registry/docker.md (hunk 4 above)
Comment (kneal) [reply content/plugins/pipeline/registry/docker.md]: I think that relative link will work but I can do the full link to be safe

Comment (wass3r) [reply content/plugins/pipeline/registry/docker.md]: yea, i don't think that will work.. i think that will reference https://go-vela.github.io/kaniko?

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**